### ISSUE TITLE
Mumble client crashes when changing application language

### ISSUE THREAD

**Issue by woodkoodaa** [mumble-voip/mumble#4932]
**Describe the bug**
Mumble client crashes when changing application language while connected to server. After "recovering" from the crash, language is changed. Problem does not happen when I am not connected to server.

**Steps to Reproduce**
Open Mumble
Connect to server
Change language in Configure - Settings - User Interface - Language, click OK
Mumble prompts to restart Mumble because of changed setting, click Yes
Mumble closes window
Mumble does not automatically restart
Try to open Mumble from Start menu
Nothing happens
(In Task Manager, Mumble is listed under Background processes)
End Task for Mumble in Task Manager
Open Mumble from Start menu
Mumble crashes (without opening any window)
Send report -> Crash uploading error code 401
After clicking OK to crash uploading error, Mumble client opens automatically
Language is changed.

**Expected behavior**
After accepting to restart Mumble to change language, I expect the application to close, automatically restart and language changed.

**Desktop (please complete the following information):**
 - OS: Windows 10 20H2 19042.928
 - Version: Mumble 1.4.0 2021-04-03 ge0d0056 snapshot (Development snapshot 4)

**Comment by Krzmbrzl** [comment]
Is this reproducible?
I just tested this on Linux and there it seems that I can change the language while being connected to a server just fine.

If this is reproducible, could you check for Mumble's process ID (https://support.kaspersky.com/us/common/windows/6325) while it is running and then after it has closed, could you check the process ID of the "background service" that is still running and preventing you from opening a new Mumble instance? Are these IDs the same or do they differ?

**Comment by woodkoodaa** [comment]
The problem is reproducible.

Starting Mumble normally: PID 1728
After clicking Yes to restart Mumble (after changing language), it is shown in background processes: PID 2244
Opening Mumble from Start menu: nothing happens
End Task in Task Manager
Open via Start menu: PID 9032
After crash and (trying to) send crash report, client opening as it should: PID 9032

**Comment by Krzmbrzl** [comment]
Okay so to me it seems like Mumble terminates properly, but then hangs somewhere during startup.

Could you perhaps try uninstalling the snapshot and installing the latest stable release (1.3.4) instead to see if the problem exists there as well?

**Comment by woodkoodaa** [comment]
Tried these versions:

No problem:
mumble-1.3.4.winx64.msi
Mumble-1.4.0 2020-12-24 gdb4b6be snapshot.winx64.msi

Problem:
Mumble-1.4.0 2021-02-23 g569336e snapshot.x64.msi
Mumble-1.4.0 2021-03-20 gf8dc8b0 snapshot.x64.msi
Mumble-1.4.0 2021-04-03 ge0d0056 snapshot.x64.msi

**Comment by Krzmbrzl** [comment]
Oh that is interesting. So the problematic change must have occurred between the release of the first and second development snapshot. In the respective changelog I can't find anything suspicious though :eyes:

**Comment by Krzmbrzl** [comment]
A user on our IRC channel was also able to reliably trigger this bug using Windows 7:
> I was able to reproduce this multiple times. I installed Mumble-1.4.0\~2021-04-03\~ge0d0056~snapshot.x64 on an up-to-date Windows 7 computer. Followed the steps. Same results. I did notice in my testing that it will not crash if not connected to a server when changing the language. I tried 4 times with each (connected/disconnected), it always crashed when connected, never crashed when disconnected.

Thus this seems to be a general problem. I think eventually this needs to be bisected to figure out which commit is causing the issue.
Alternatively someone more skilled with Windows development than me (e.g. @davidebeatrici ?) could attach a debugger to the application to figure out where it ends up hanging.

**Comment by davidebeatrici** [comment]
Cannot reproduce. What's the language you're trying to switch to?

**Comment by woodkoodaa** [comment]
I switched between "system default" which is english-us and suomi (fi) (finnish).

**Comment by davidebeatrici** [comment]
Unfortunately I still cannot reproduce the issue.

**Comment by Krzmbrzl** [comment]
@davidebeatrici are you using a self-compiled version or one of our released binaries directly?

**Comment by davidebeatrici** [comment]
I tested both.

**Comment by Krzmbrzl** [comment]
@davidebeatrici what is your system's language?

**Comment by davidebeatrici** [comment]
English (US).

**Comment by Krzmbrzl** [comment]
Okay so that's the same. Hm let me see if I can reproduce this issue in my VM...

**Comment by Krzmbrzl** [comment]
I am indeed able to reproduce it there. Now I only have to find the time to bisect the issue. I can't tell yet when that will happen though xD

**Comment by davidebeatrici** [comment]
Please note that I have UTF-8 support enabled:

![](https://user-images.githubusercontent.com/5897523/115156399-6248d680-a084-11eb-9aee-9da40a113eb0.png)

Aside from that, I only have a few non-default locale settings:

![](https://user-images.githubusercontent.com/5897523/115156403-6a087b00-a084-11eb-8be3-a5cfebbbd29f.png)

**Comment by ViBE-HU** [comment]
i noticed similar issue. i have an old ThinkPad and installed Windows 10 a few months ago. it's an up to date system. first i tested with the stable 1.3.4 build. it runs without any issue. yesterday i removed it and started to test the latest snapshot and update the translation. but this build does not start. it crashes instantly. the system log says that the program crashes because of the opus.dll cause maybe it a filesystem issue. but it's not. then i removed the opus.dll and now the program crashes because of the mumble_app.dll.

**Comment by Krzmbrzl** [comment]
@ViBE-HU I don't see how your issue is similar to the one described here :thinking:

In any case though it would be helpful if you could provide the actual crash message. Otherwise it will be hard to figure out what the issue is...

**Comment by Krzmbrzl** [comment]
> if this feature works

Unfortunately our backend for that is currently broken xD

That log you provided is the one _after_ you have deleted `opus.dll`, isn't it? Could you give me the one with this DLL still present?

And if you don't have an argument for why this is in fact the same issue as the one originally described in this issue, I think it would be best if we moved our discussion into a separate issue. wdyt?

**Comment by ViBE-HU** [comment]
yeah, i remembered that i already tried and somehow it does nothing when i sent the report back in the days.

yes if it's not related i can open another issues. i just don't wanted to make duplicates. here's the opus.dll related Windows event log. first of all if that dll is present an event with error code 1005 created. it is a long log what describe that the filesystem maybe compromised and should check with CHKDSK. but it's totally irrelevant. then the other event 1000:

```
A hibát okozó alkalmazás neve: mumble.exe, verzió: 1.4.0.191, időbélyeg: 0x60a164a5
A hibát okozó modul neve: opus.dll, verzió: 0.0.0.0, időbélyeg: 0x60a16375
Kivételkód: 0xc000001d
Hiba pozíciója: 0x000000000001f350
A hibát okozó folyamat azonosítója: 0x17ec
A hibát okozó alkalmazás indításának időpontja: 0x01d789245c117909
A hibát okozó alkalmazás elérési útja: C:\Program Files\Mumble\client\mumble.exe
A hibát okozó modul elérési útja: C:\Program Files\Mumble\client\opus.dll
Jelentés azonosítója: ef84c957-293f-4e96-82b0-72ce94f17bb0
A hibát okozó csomag teljes neve: 
A hibát okozó csomag relatív alkalmazásazonosítója: 
```

**Comment by Hartmnt** [comment]
More likely than not fixed by #6466 

If any future reader stumbles across this and this is still a problem in Mumble 1.6.x, please open a new issue
